### DISCUSSION TITLE
How to configure proxy in zed settings?

### DISCUSSION THREAD

**Discussion by ssarraju** [zed-industries/zed#7525] [Feature Requests]
How to configure proxy in zed settings?

**Comment by d1y** [comment]
FYI:
```bash
http_proxy=http://192.168.1.128:1085 https_proxy=http://192.168.1.128:1085 zed project
```
https://github.com/zed-industries/zed/pull/6765
https://github.com/zed-industries/zed/pull/6765/files

**Reply by SomeoneToIgnore** [reply]
So looks like we need a docs entry?
People would not be able to find and read PRs that easy to fetch that info.

**Comment by ZTfer** [comment]
add `"proxy" : "socks5://localhost:10808"` or `"proxy" : "http://127.0.0.1:10809"` in `.configs/zed/settings.json` file, then press `cmd + q` to quit zed app, restart zed app, the proxy will work immediately.

**Reply by Rik002** [reply]
how to include https proxy and moreover, network also requires username and password.

**Reply by ZTfer** [reply]
try this?
`"http://userName:yourPassword@yourProxyURL.com:8080"`

**Reply by ZTfer** [reply]
I found some information:
https://github.com/TooTallNate/proxy-agents/tree/main/packages/proxy-agent
https://askubuntu.com/questions/82880/how-do-i-set-a-system-wide-proxy-with-a-username-and-password
https://serverfault.com/questions/1103523/how-do-i-set-http-proxy-and-https-proxy-with-the-username-and-password-with-it-i

Without username & passwords:
`"proxy" : "socks5://yourProxyURL.com:8080"`
`"proxy" : "http://yourProxyURL.com:8080"`

With username & passwords:
`"proxy" : "socks5://username:password@yourProxyURL.com:8080"`
`"proxy" : "http://username:password@yourProxyURL.com:8080"`

**Reply by Rik002** [reply]
doesn't work for me, I don't know, tried various combinations. But doesn't seem to change the situation. Moreover, I have http and https for my uni proxy

**Comment by 1WorldCapture** [comment]
btw, you could use shortcut "ctrl + ," or "CMD + ," to open the settings.json, and add "proxy" config to it, mentioned above

**Comment by xiaguangbo** [comment]
settings.json
```json
{
  "ui_font_size": 16,
  "buffer_font_size": 16,
  "theme": {
    "mode": "system",
    "light": "One Light",
    "dark": "One Dark"
  },
  "proxy" : "http://192.168.1.13:7890"
}
```

**Comment by Set27** [comment]
When I use `http` proxy I have error with npm.
Socks + password does not work.

**Comment by desprit** [comment]
I have it as
```
"proxy": "http://username:password@ip:port"
```
and when I restart zed NPM stops working + copilot is not able to perform github signin.

p.s. the same setup works in vscode

**Reply by Set27** [reply]
What type of proxy do u use?

**Comment by thynson** [comment]
FYI: at least on macOS, zed prefers to resolve `localhost` to `::1` instead of `127.0.0.1`, so some proxy software only listen for ipv4 localhost.

**Comment by BLumia** [comment]
The `proxy` setting seems will not apply for:

- Agents that spawned by zed (e.g. `claude-code` and `gemini` cli)
- `npm` command spawned by zed (at least when using built-in `npm` instead of the system one)

Hope we can have some configuration to make these cases support proxy as well.

**Comment by wujiahao15** [comment]
Can zed remote use proxy? I want to make sure that my extensions can be downloaded successfully in remote mode.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
